src/main/java/fr/insee/survey/datacollectionmanagement/questioning/controller/SurveyUnitController.java

```diff
@@ -60,33 +70,44 @@ public Page<SurveyUnitDto> getSurveyUnits(
     @Operation(summary = "Multi-criteria search survey-unit")
     @GetMapping(value = Constants.API_SURVEY_UNITS_SEARCH, produces = "application/json")
     @ApiResponses(value = {
-            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = SurveyUnitPage.class))),
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(array = @ArraySchema(schema = @Schema(implementation = SearchSurveyUnitDto.class)))),
             @ApiResponse(responseCode = "404", description = "Not found"),
             @ApiResponse(responseCode = "400", description = "Bad Request")
     })
-    public Page<SurveyUnitDto> searchSurveyUnits(
-            @RequestParam(required = false) String idSu,
-            @RequestParam(required = false) String identificationCode,
-            @RequestParam(required = false) String identificationName,
+    public Page<SearchSurveyUnitDto> searchSurveyUnits(
+            @RequestParam(required = true) String searchParam,
+            @RequestParam(required = true) @Valid @ValidSurveyUnitParam @Schema(description = "id or code or name")String searchType,
```

Comment:
It should be working without custom annotations here by giving directly the enum as parameter

Reply:
Same goes for the other custom validator annotation